app/src/org/commcare/android/logging/ForceCloseLogSerializer.java

```diff
@@ -0,0 +1,100 @@
+package org.commcare.android.logging;
+
+import org.commcare.android.database.SqlStorage;
+import org.javarosa.core.log.LogEntry;
+import org.javarosa.core.log.StreamLogSerializer;
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.util.SortedIntSet;
+import org.xmlpull.v1.XmlSerializer;
+
+import java.io.IOException;
+import java.util.Hashtable;
+
+/**
+ * Convert xpath error logs to xml.
+ *
+ * @author Aliza Stone
+ */
+public class ForceCloseLogSerializer extends StreamLogSerializer implements DeviceReportElement {
+
+    private ForceCloseLogEntry singleEntry;
+    private SqlStorage<ForceCloseLogEntry> logStorage;
+    private XmlSerializer serializer;
+
+    public ForceCloseLogSerializer(ForceCloseLogEntry entry) {
+        this.singleEntry = entry;
+    }
+
+    public ForceCloseLogSerializer(final SqlStorage<ForceCloseLogEntry> logStorage) {
+        this.logStorage = logStorage;
+
+        this.setPurger(new Purger() {
+            @Override
+            public void purge(final SortedIntSet IDs) {
+                logStorage.removeAll(new EntityFilter<LogEntry>() {
+                    public int preFilter(int id, Hashtable<String, Object> metaData) {
+                        return IDs.contains(id) ? PREFILTER_INCLUDE : PREFILTER_EXCLUDE;
+                    }
+
+                    public boolean matches(LogEntry e) {
+                        throw new RuntimeException("can't happen");
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    public void writeToDeviceReport(XmlSerializer serializer) throws IOException {
+        this.serializer = serializer;
+
+        serializer.startTag(DeviceReportWriter.XMLNS, "force_close_subreport");
```

Comment:
This means we're going to start missing these in the short term, right? Until we get HQ to start parsing these?